src/codegen/abi.js

```diff
@@ -59,18 +60,60 @@ module.exports = class AbiCodeGenerator {
             name = `param${index}`
           }
           paramsClass.addMethod(
-            tsCodegen.method(
+            input.get('type') === 'tuple' ?
+              tsCodegen.method(
+                `get ${name}`,
+                [],
+                tsCodegen.namedType(name).capitalize(),
```

Comment:
I'd move the `tsCodegen.namedType(name).capitalize()` into a `tupleClassName` variable, so you can reuse it in the generated `return` statement below.

Reply:
Another thought: we can't name tuples after their parameter names, that'll cause too many naming clashes.

Let's use event name (captialized) + param name or index as the name for the tuple classes.

Reply:
Maybe even append "Struct" to it, so e.g. `TransferParam0Struct` or `TransferParamFooStruct`.